libmproxy/proxy/root_context.py

```diff
@@ -33,6 +59,7 @@ def __init__(self, client_conn, config, channel):
         self.client_conn = client_conn
         self.channel = channel
         self.config = config
+        self._client_tls = False
```

Comment:
Is this just for caching the magic check? I think we should avoid additional state as much as possible.